src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        values[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        grow();
+        if (index != size) {
+            checkIndexWithSameSize(index);
+            System.arraycopy(values, index, values, index + 1, size - index);
+        }
+        values[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        T[] a = (T[]) new Object[list.size()];
+        for (int i = 0; i < list.size(); i++) {
+            a[i] = list.get(i);
+        }
+        int varLength = a.length;
+        grow();
+        System.arraycopy(a,0,values,size, varLength);
+        size += varLength;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexWithSameSize(index);
+        return values[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndexWithSameSize(index);
+        values[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexWithSameSize(index);
+        final T removedElement = values[index];
+        System.arraycopy(values, index + 1,
+                values, index,
+                values.length - index - 1);
+        values[values.length - 1] = null;
```

Comment:
I think you wanted to do the next
```suggestion
        values[--size] = null;
```
However, it's not necessary